addons/project/views/project_views.xml

```diff
@@ -1178,13 +1163,32 @@
                         </group>
                         <group>
                             <field name="active" invisible="1"/>
-                            <field name="partner_id" widget="res_partner_many2one" class="o_task_customer_field" attrs="{'invisible': [('project_id', '=', False)]}"/>
+                            <field name="partner_id" nolabel="0" widget="res_partner_many2one" class="o_task_customer_field" attrs="{'invisible': [('project_id', '=', False)]}"/>
                             <field name="partner_phone" widget="phone" attrs="{'invisible': True}"/>
-                            <field name="date_deadline" attrs="{'invisible': [('is_closed', '=', True)]}"/>
-                            <field name="legend_blocked" invisible="1"/>
-                            <field name="legend_normal" invisible="1"/>
-                            <field name="legend_done" invisible="1"/>
+                            <label for="date_deadline"/>
+                            <div id="date_deadline_and_recurring_task" class="d-inline-flex w-100">
+                                <field name="date_deadline" nolabel="1"/>
+                                <field name="recurring_task" nolabel="1" class="ms-0" style="width: fit-content;"
+                                       widget="boolean_icon" options="{'icon': 'fa-repeat'}"
+                                       attrs="{'invisible': ['|', ('allow_recurring_tasks', '=', False), ('active', '=', False)]}"
+                                       groups="project.group_project_recurring_tasks"
+                                />
+                            </div>
+                            <label for="repeat_interval" groups="project.group_project_recurring_tasks" attrs="{'invisible': [('recurring_task', '=', False)]}" />
+                            <div attrs="{'invisible': [('recurring_task', '=', False)]}" class="d-flex" groups="project.group_project_recurring_tasks">
+                                <field name="repeat_interval" attrs="{'required': [('recurring_task', '=', True)]}"
+                                       class="me-2" style="max-width: 2rem !important;" />
+                                <field name="repeat_unit" attrs="{'required': [('recurring_task', '=', True)]}"
+                                       class="me-2" style="max-width: 4rem !important;" />
+                                <field name="repeat_type" attrs="{'required': [('recurring_task', '=', True)]}"
+                                       class="me-2" style="max-width: 15rem !important;" />
+                                <field name="repeat_until" attrs="{'invisible': [('repeat_type', '!=', 'until')], 'required': [('repeat_type', '=', 'until')]}"
+                                       class="me-2" />
```

Comment:
to check if we could not find another solution then adding style. 

TODO for me. 😇